boa/contracts/abi/contract.py

```diff
@@ -48,7 +54,7 @@ def marshal_to_python(self, computation, abi_type: list[str]) -> tuple[Any, ...]
         :param computation: the computation object returned by `execute_code`
         :param abi_type: the ABI type of the return value.
         """
-        if computation.is_error:
+        if computation.is_error or (abi_type and not computation.output):
```

Comment:
i don't really like the truthy uses here. actually it seems the issue is that abi decoding could fail, right? can we skip this check and then just catch the exception from `abi_decode()` a few lines down? that would handle other abi decoding errors too, not sure if that's a good thing or not.

Reply:
This is to raise proper errors when there is no bytecode deployed in that address.
This test: https://github.com/vyperlang/titanoboa/blob/2ca7e30d0a552b6a1c4274d31b5a76480a16ed49/tests/integration/fork/test_abi_contract.py#L51

If we don't do this here, you get a `DecodeError: Error decoding '0x' as '(uint8)' - Value length is not the expected size of 32 bytes` - which gives no information to the user.
With this check the error changes to `BoaError:   (unknown location in <crvusd_abi.json interface at 0x0000000000000000000000000000000000000000 (WARNING: no bytecode at this address!)> crvusd_abi.decimals() -> ['uint8'])`

I think using truthy here makes sense, since it doesn't matter whether `abi_type`/`computation.output` are `None` or `''`